sdk/src/Services/S3/Custom/Model/Internal/MarshallTransformations/CopyObjectRequestMarshaller.cs

```diff
@@ -73,15 +73,17 @@ public IRequest Marshall(CopyObjectRequest copyObjectRequest)
 
             if (copyObjectRequest.IsSetUnmodifiedSinceDateUtc())
                 request.Headers.Add(HeaderKeys.XAmzCopySourceIfUnmodifiedSinceHeader, S3Transforms.ToStringValue(copyObjectRequest.UnmodifiedSinceDateUtc.Value));
-
-            if (copyObjectRequest.IsSetTagSet())
-            {
-                request.Headers.Add(S3Constants.AmzHeaderTagging, AmazonS3Util.TagSetToQueryString(copyObjectRequest.TagSet));
-                request.Headers.Add(S3Constants.AmzHeaderTaggingDirective, TaggingDirective.REPLACE.Value);
-            }
-            else
+            
+            if (copyObjectRequest.IsSetTaggingDirective())
             {
-                request.Headers.Add(S3Constants.AmzHeaderTaggingDirective, TaggingDirective.COPY.Value);
+                if (copyObjectRequest.TaggingDirective == TaggingDirective.REPLACE)
```

Comment:
I thought of something similar but this could lead to adding a value of the same key, for example if `copyObjectRequest.TaggingDirective == TaggingDirective.Copy`, then we would do 
`request.Headers.Add(S3Constants.AmzHeaderTaggingDirective, copyObjectRequest.TaggingDirective.Value);` 

AND 
`request.Headers.Add(S3Constants.AmzHeaderTaggingDirective, TaggingDirective.COPY.Value);` which would lead to an exception